doc/specs/stdlib_stats.md

```diff
@@ -166,6 +178,75 @@ program demo_mean
 end program demo_mean
 ```
 
+## `median` - median of array elements
+
+### Status
+
+Experimental
+
+### Description
+
+Returns the median of all the elements of `array`, or of the elements of `array` along dimension `dim` if provided, and if the corresponding element in `mask` is `true`.
+
+After that the elements are sorted in an increasing order, e.g. `array_sorted =
+sort(array)`, the median of the elements of `array` are defined as, if
+`n = size(array)` is an even number:
+
+```
+median(array) = array_sorted( floor( (n + 1) / 2.))
+```
+
+or as if `n` is an odd number:
+
+```
+median(array) = mean( array_sorted( floor( (n + 1) / 2.):floor( (n + 1) / 2.) + 1 ) )
+```
+
+The array is sorted using the subroutine `sort` provided by the `stdlib_sorting`
+module.
+
+### Syntax
+
+`result = [[stdlib_stats(module):median(interface)]](array [, mask])`
+
+`result = [[stdlib_stats(module):median(interface)]](array, dim [, mask])`
+
+### Class
+
+Generic subroutine
+
+### Arguments
+
+`array`: Shall be an array of type `integer` or  `real`.
```

Comment:
Here could you add a comment about the range of supported kinds for each?

Reply:
Not sure what you  mean. I copied the other descriptions in these specs.

Reply:
Oh I mean like single precision, double precision, etc. If that is already implied by the statement above, then no problem. I get confused because in Fortran "real" is single precision, but we can also speak of the "real kinds".